### PR TITLE
SLCORE-326 Allow embedded plugins working in connected mode

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: core/src/test/java/org/sonarsource/sonarlint/core/mediumtest/ConnectedExtraPluginMediumTest.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        + "  public void foo() {\n"
+        + "    int x;\n"
+        + "    System.out.println(\"Foo\");\n"
+        + "    System.out.println(\"Foo\"); //NOSONAR\n"
+        + "  }\n"
+        + "}",
+      false);
+  }
+
+  private ClientInputFile prepareJavaTestInputFile() throws IOException {
```
Comment (henryju) [inline core/src/test/java/org/sonarsource/sonarlint/core/mediumtest/ConnectedExtraPluginMediumTest.java:205]: Dead code?

**Review by henryju — APPROVED** [review]
Minor feedback then LGTM. I think you should wait after release of SLCORE 6.1 before merging this.

File: core/src/test/java/org/sonarsource/sonarlint/core/mediumtest/ConnectedExtraPluginMediumTest.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+      tuple("java:S113", null, inputFile.getPath(), "MINOR"),
+      tuple("java:S1228", null, null, "MINOR"),
+      tuple("java:S1106", 1, inputFile.getPath(), "MINOR"),
+      tuple("java:S1106", 2, inputFile.getPath(), "MINOR"),
+      tuple("java:S1451", null, inputFile.getPath(), "BLOCKER"),
+      tuple("java:NoSonar", 5, inputFile.getPath(), "MAJOR"));
+  }
+
+  @SonarLintSide(lifespan = "MODULE")
+  static class FakeModuleFileListener implements ModuleFileListener {
```
[the remark this answers is not in the document]
Comment (kirill-knize-sonarsource) [reply core/src/test/java/org/sonarsource/sonarlint/core/mediumtest/ConnectedExtraPluginMediumTest.java]: Removed.

File: core/src/test/java/org/sonarsource/sonarlint/core/mediumtest/ConnectedExtraPluginMediumTest.java (hunk 1 above)
Comment (kirill-knize-sonarsource) [reply core/src/test/java/org/sonarsource/sonarlint/core/mediumtest/ConnectedExtraPluginMediumTest.java]: Removed.

**Pull request merged**